### PR TITLE
[jnigen] FixCI

### PR DESCRIPTION
**Pull request by HosseinYousefi**
Fix a bug where the package would either be fully resolved by sources or classes but not by both.
Fix CI.

### CODE REVIEW

File: pkgs/jnigen/java/src/main/java/com/github/dart_lang/jnigen/apisummarizer/disasm/AsmClassVisitor.java [hunk 1]
```diff
@@ -64,14 +63,11 @@ public void visit(
...
-        visited.stream()
-            .filter(decl -> decl.binaryName.equals(binaryName))
-            .collect(Collectors.toList());
-    // If the order of visit is outer first inner second.
-    // We still want to correct the modifiers.
-    if (!alreadyVisitedInnerClass.isEmpty()) {
-      alreadyVisitedInnerClass.get(0).modifiers = TypeUtils.access(access);
+
+    if (visited.containsKey(binaryName)) {
+      // If the order of visit is outer first inner second.
```
Comment (dcharkes) [inline pkgs/jnigen/java/src/main/java/com/github/dart_lang/jnigen/apisummarizer/disasm/AsmClassVisitor.java:68]: Nit, sentence: If ..., then ... ?

File: pkgs/jnigen/java/src/main/java/com/github/dart_lang/jnigen/apisummarizer/disasm/AsmClassVisitor.java [hunk 2]
```diff
@@ -64,14 +63,11 @@ public void visit(
...
+      // We still want to correct the modifiers.
```
Comment (dcharkes) [inline pkgs/jnigen/java/src/main/java/com/github/dart_lang/jnigen/apisummarizer/disasm/AsmClassVisitor.java:69]: Nit: we

File: pkgs/jnigen/java/src/main/java/com/github/dart_lang/jnigen/apisummarizer/disasm/AsmClassVisitor.java [hunk 3]
```diff
@@ -23,11 +22,11 @@ private static Param param(
     return param;
   }
 
-  public List<ClassDecl> getVisited() {
+  public Map<String, ClassDecl> getVisited() {
     return visited;
   }
 
-  List<ClassDecl> visited = new ArrayList<>();
+  Map<String, ClassDecl> visited = new HashMap<>();
```
Comment (dcharkes) [inline pkgs/jnigen/java/src/main/java/com/github/dart_lang/jnigen/apisummarizer/disasm/AsmClassVisitor.java:29]: I was going to write `... visited = {};`, oh wait, this is Java ....

File: pkgs/jnigen/java/src/main/java/com/github/dart_lang/jnigen/apisummarizer/Main.java [hunk 4]
```diff
@@ -137,14 +122,16 @@ public static void main(String[] args) throws FileNotFoundException {
         JsonWriter.writeJSON(AsmSummarizer.run(classStreamProviders), output);
         break;
       case AUTO:
-        List<ClassDecl> decls = new ArrayList<>();
-        if (!sourceFiles.isEmpty()) {
-          decls.addAll(runDoclet(javaDoc, sourceFiles, options));
-        }
+        Map<String, ClassDecl> classes = new HashMap<>();
+        // Preferring DOCLET as the source of summary a class exists in
```
Comment (dcharkes) [inline pkgs/jnigen/java/src/main/java/com/github/dart_lang/jnigen/apisummarizer/Main.java:126]: We should probably add a test that exercises this.

File: pkgs/jnigen/lib/src/elements/elements.dart [hunk 5]
```diff
@@ -33,13 +33,9 @@ class Classes implements Element<Classes> {
...
   final Map<String, ClassDecl> decls;
 
-  factory Classes.fromJson(List<dynamic> json) {
-    final decls = <String, ClassDecl>{};
-    for (final declJson in json) {
-      final classDecl = ClassDecl.fromJson(declJson);
-      decls[classDecl.binaryName] = classDecl;
-    }
-    return Classes(decls);
+  factory Classes.fromJson(Map<String, dynamic> json) {
```
Comment (dcharkes) [inline pkgs/jnigen/lib/src/elements/elements.dart:36]: Apparently this is not covered with tests according to the coverage bot?

**Review by dcharkes — APPROVED** [review]
lgtm w comments

Thanks @HosseinYousefi !

**Comment by mosuem** [comment]
> Health / breaking changes bot is unhappy.
> Is it misconfigured?

I think this is a bug in the `dart_apitool` - I filed a [bug there](https://github.com/bmw-tech/dart_apitool/issues/177).

File: pkgs/jnigen/lib/src/elements/elements.dart (hunk 5 above)
Comment (HosseinYousefi) [reply pkgs/jnigen/lib/src/elements/elements.dart]: It should be, we don't have coverage link anymore? We used to get them from codecov bot in jnigen repo. @mosuem Could we set that up here as well?

Comment (HosseinYousefi) [reply pkgs/jnigen/lib/src/elements/elements.dart]: Nevermind, the tests were not all passed before.

**Pull request merged**